jme3-core/src/main/java/com/jme3/scene/Geometry.java

```diff
@@ -136,6 +136,24 @@ public Geometry(String name, Mesh mesh) {
         this.mesh = mesh;
     }
 
+    /**
+     * Create a geometry node with mesh data and material.
+     *
+     * @param name The name of this geometry
+     * @param mesh The mesh data for this geometry
+     * @param material The material for this geometry
+     */
+    public Geometry(String name, Mesh mesh, Material material) {
```

Comment:
This new constructor should call 
` public Geometry(String name, Mesh mesh) `
instead of copying its code.

Reply:
Totally right about constructor code duplication. To be changed
Regarding setMaterial as method of Spatial, it is true. Idea here is to have a fully functional Geometry instance , as soon as the constructor is called.
Let's hear from others.